lua/telescope/_extensions/file_browser/make_entry.lua

```diff
@@ -185,6 +188,21 @@ local make_entry = function(opts)
     end
     if k == "stat" then
       local stat = vim.loop.fs_stat(t.value)
+      if not stat then
+        -- Kinda ick but since stat is a table of tables (and stuff),
+        -- we have to account for those subtables
+        log.warn("Unable to get stat for " .. t.Path:absolute())
```

Comment:
Just as a side note `t.value` should always be the absolute path already, so should not be needed to infer via `Path`.

Makes me think maybe I should get rid of `Path` where possible. Let's see, quite some work for probably unnoticably better latency :sweat_smile:

Reply:
I think removing path will actually help out quite a bit with performance, plenary does alot under the hood here. 

I tend to see a minimum of 3 duplicate calls to a single libuv function because of it, and plenary calls several with each time they do something. 

That said, that is likely a separate work effort (if you want, feel free to open a new issue and assign it to me)

Reply:
> I think removing path will actually help out quite a bit with performance

Improve performance with regards to what? At most, it should increase latency for up to 1000 initially visible entries (gut feeling tells me 5-10ms) as otherwise `Path` creation is fully deferred and should be appropriately cached within a single instance. Of course it can be improved, but the improvement would be unnoticeable in almost all usage scenarios.

Reply:
I'll have to poke this a bit more this weekend, but I noticed that `Path` is a bit slow due to all the wrapping around it. Latency (response time I suppose?) will improve (decrease) as you suggested.
However `telescope-file-browser` is clever and caches (smart!) to help avoid this. Still, I also wonder if removing `Path` makes sense. That said, `telescope-file-browser` uses it for a decent amount of filesystem interfacing _and_ to do logic (even via assumption), so I don't think its a small work effort. Maybe something to explore after a good profiling of the plugin? It might not be the best place to target for optimization (work vs reward)

Reply:
Huh, looks like I've got it right haha

```lua
local bench = require "plenary.benchmark"
local Path = require "plenary.path"

local filename = vim.api.nvim_buf_get_name(0)

bench("Path", {
  warmup = 10,
  runs = 50,
  fun = {
    {
      "Path creation",
      function()
        for _ = 1, 1000 do
          Path:new(filename)
        end
      end,
    },
  },
})

```

Benchmark Group: 'Path' -----------------------                                                                                                                                                 
Benchmark #1: 'Path creation'                                                                                                                                                                   
  Time(mean ± σ):      9.4 ms ±   2.8 ms                                                                                                                                                        
  Range(min … max):    7.7 ms …  18.1 ms  50 runs

Which is practically speaking the maximum latency that can be incurred from Path creation during finding as of now.

cc @Conni2461 another good usage of `plenary.benchmark` :)